### DISCUSSION TITLE
Why does returning plain object from a loader work?

### DISCUSSION THREAD

**Discussion by freddie-freeloader** [remix-run/remix#9837] [Q&A] [answered]
When using loaders in Remix, one should -- according to the documentation -- return a response object:

```typescript
export async function loader() {
  const users = await db.users.findMany();
  const body = JSON.stringify(users);
  return new Response(body, {
    headers: {
      "Content-Type": "application/json",
    },
  });
}
```

There is also a helper function `json`, that simplifies the code:

```typescript
import { json } from "@remix-run/node"; // or cloudflare/deno

export const loader = async () => {
  const users = await fakeDb.users.findMany();
  return json(users);
};
```

By mistake, I forgot to wrap the returned object in a response:

```typescript
export const loader = async () => {
  const users = await fakeDb.users.findMany();
  return users;
};
```

I got no type error though, and the loader worked just fine with `useLoaderData<typeof loadRouteData>()`.

I assume, that the object is automatically serialized and wrapped in a `Response` object(?).

Is this a behavior I can rely on? Or is this considered bad practice and I should wrap the returned object in `json` like in the documentation?

**Comment by sergiodxa** [comment, marked as answer]
Remix is doing `json(loaderReturnData)` for you if you don't return a Response.

**Reply by freddie-freeloader** [reply]
Thank you for the quick reply 😊
